resources/kubernetes/operator/helm/values.yaml

```diff
@@ -95,6 +95,11 @@ collectors:
               action: insert
             - key: app.label.version
               action: delete
+        resource/hostname:
+          attributes:
+            - key: host.name
+              from_attribute: k8s.node.name
+              action: upsert
```

Comment:
Is this something the `k8sattributes` should do by default? If so, can we propose these changes upstream?

Reply:
That could be an option, similar to the `override` functionality in the `resource detector processor` (e.g. overrides the host.name value with the cloud instance ID). But I have two concerns:

- Upstream agreement with `host.name == k8s.node.name`: looks like a custom workaround to fix the Infrastructure UI relationships. From the container perspective, it has its own host.name value, which is actually used among the cluster for communication.
- Otlp data that already contains `k8s` resource attributes that won't be processed by the k8sattributes processor.

Reply:
> Upstream agreement with host.name == k8s.node.name: looks like a custom workaround to fix the Infrastructure UI relationships. From the container perspective, it has its own host.name value, which is actually used among the cluster for communication.

IMHO, that is a symptom of `host.name` in SemConv being underspecified. In particular, there's no clear statement about the semantics of the `host.name` in containerized environments. 
In particular it says `... On Unix systems, it may contain what the hostname command returns, or the fully qualified hostname ...`. For containerized system that means that SDKs will report that value which from the perspective from within a container is per default the pod name. I'm not sure if that behavior is really intended by SemConv or just not thought through well enough. The consequence is, that `host.name` is not a reliable attribute with that. So I don't think it's an Elastic-specific challenge.

Reply:
I guess we need to understand all scenarios of the host.name population per different  k8s setup. Created internal issue to summarise all the above